src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonServiceIntTest.kt

```diff
@@ -25,6 +25,39 @@ class PersonServiceIntTest : MessagingMultiNodeTestBase() {
     personRepository.deleteAll()
   }
 
+  @Test
+  fun `should find candidate records only in searching source system`() {
+    val personToFind = Person(
+      otherIdentifiers = OtherIdentifiers(pncIdentifier = PNCIdentifier.from("2003/0011985X")),
+      sourceSystemType = SourceSystemType.LIBRA,
+    )
+    createPerson(personToFind)
+    createPerson(
+      Person(
+        otherIdentifiers = OtherIdentifiers(pncIdentifier = PNCIdentifier.from("1981/0154257C")),
+        sourceSystemType = SourceSystemType.HMCTS,
+      ),
+    )
+    createPerson(
+      Person(
+        otherIdentifiers = OtherIdentifiers(pncIdentifier = PNCIdentifier.from("1981/0154257C")),
+        sourceSystemType = SourceSystemType.NOMIS,
+      ),
+    )
+    createPerson(
+      Person(
+        otherIdentifiers = OtherIdentifiers(pncIdentifier = PNCIdentifier.from("1981/0154257C")),
+        sourceSystemType = SourceSystemType.DELIUS,
+      ),
+    )
+
+    val personEntities = personService.findCandidateRecords(personToFind)
```

Comment:
If I understand this correctly, the search would never match any of the other records anyway because PNC is different. It might be an idea to make the records look more similar by supplying a bit more data, such that the test fails if you omit the sourceSystem from the comparison because the records match.

I may well be wrong but that is how the test appears to me

Reply:
Yeah thats why you never copy and paste a test, ill update it